Sources/Error Handling/Assertions.swift

```diff
@@ -44,3 +44,21 @@ func RCTestAssertNotMainThread(
     )
     #endif
 }
+
+@inline(__always)
+func RCTestAssertMainThread(
+    function: StaticString = #function,
+    file: StaticString = #file,
+    line: UInt = #line
+) {
+    #if DEBUG
+    if ProcessInfo.isRunningIntegrationTests {
```

Comment:
Only running the assertions when running integration tests, because on unit tests a lot of the dependencies are mocked, and mocks do not always execute completions on the main thread causing false positives.